File: 프로그래머스/2/86971. 전력망을 둘로 나누기/전력망을 둘로 나누기.py
def solution(n, wires):
    answer = n+1
    for i in range(len(wires)):
        cnt = 0
        used = [0] * len(wires)
        used[i] = 1
        conn = []
        conn.append(wires[i][0])
        while conn:
            st = conn.pop()
            for j in range(len(wires)):
                if used[j] == 1: continue
                if wires[j][0] == st:
                    conn.append(wires[j][1])
                    used[j] = 1
                    cnt += 1
                elif wires[j][1] == st:
                    conn.append(wires[j][0])
                    used[j] = 1
                    cnt += 1
        answer = min(answer, abs(n - 2 - cnt - cnt))
    return answer

File: 프로그래머스/2/86971. 전력망을 둘로 나누기/전력망을 둘로 나누기.js
function solution(n, wires) {
    let answer = n + 1;
    
    for (let i = 0; i < wires.length; i++) {
        let cnt = 0;
        let used = new Array(wires.length).fill(0);
        used[i] = 1;
        let conn = [wires[i][0]];
        
        while (conn.length > 0) {
            let st = conn.pop();
            for (let j = 0; j < wires.length; j++) {
                if (used[j] === 1) continue;
                if (wires[j][0] === st) {
                    conn.push(wires[j][1]);
                    used[j] = 1;
                    cnt += 1;
                } else if (wires[j][1] === st) {
                    conn.push(wires[j][0]);
                    used[j] = 1;
                    cnt += 1;
                }
            }
        }
        
        answer = Math.min(answer, Math.abs(n - 2 - cnt - cnt));
    }
    
    return answer;
}